### PR TITLE
CT-2640 Add ECR Docker image deletion job to production

### PR DESCRIPTION
**Pull request by mseedat-moj**
## Description
Add recommended Cloud Platform Docker Image deletion script to prevent buildup of unused images in the Cloud Platform registry
 
## Self-review checklist

* [x] (1) Quick stakeholder demo done OR
* [x] (2) ...bug with before and after screenshots
* [X] (3) Tests passing
* [x] (4) Branch ready to be merged (not work in progress)
* [X] (5) No superfluous changes in diff
* [X] (6) No TODO's without new ticket numbers
 
### Screenshots
N/A
 
### Related JIRA tickets
https://dsdmoj.atlassian.net/browse/CT-2640
 
### Deployment
Deploy as part of standard deployment procedure to PRODUCTION
 
### Manual testing instructions
Check current docker images in the Cloud Platform registry by executing in your terminal:

```
 aws ecr list-images --region eu-west-2 --repository-name correspondence/track-a-query-ecr  --filter "tagStatus=UNTAGGED" --query 'imageIds[*]' --output json
```

it should be empty if the cron job ran successfully

### Validation
1. To test Non-master images:
(a) copy the output and search for the word "master" - should be none

(b) Use this javascript here: http://www.jsonquerytool.com/
// Show proper dates
input.imageDetails.map((img) => { 
  const epoch = img.imagePushedAt;
  let d = new Date(0);
  d.setUTCSeconds(epoch);
  img.imagePushedAtFormatted=d;
  return img;
})

// Show any images that are more than 2 weeks old
input.imageDetails.map((img) => {
  	let d = new Date();
  	d.setDate(d.getDate() - 14)
  	d = d.getTime() / 1000
  console.log(d);
  console.log(img.imagePushedAt)
	if (img.imagePushedAt > d) return img; 
})
 - Should be all nulls (try flexing "14" to get some output)

2 - To test master images:
(a) Find last 10 master image tags
(b) Copy the Master output and search for them - You shouldn't find them
(c) Scroll through and try to find any non master images.

### Raw Output:

Note: Following is output of master and non master images that will be deleted once the delete command is included. They need to be validated some how that they fit our criteria. I did this validation locally but not reliable because this is the real cron job now, so needs cross-referencing some how. Or the Epoch dates need to show up as real dates somehow so we can check them easily.

"Total Master Images to delete":100
{
   "imageDetails":[
      {
         "registryId":"754256621582",
         "repositoryName":"correspondence/track-a-query-ecr",
         "imageDigest":"sha256:6284ad5ec8eab68536ecd13c436d2c41c78f46a8e1383361d7458c5c060a70f2",
         "imageTags":[
            "cts-master-c393d4f2"
         ],
         "imageSizeInBytes":1050756116,
         "imagePushedAt":1574424263
      },
      { 
         "registryId":"754256621582",
         "repositoryName":"correspondence/track-a-query-ecr",
         "imageDigest":"sha256:e05e77f6180c1807f5b350f74c550cca21bd8355bd03fc0644176e7285f68638",
         "imageTags":[
            "cts-master-d9981b95"
         ],
         "imageSizeInBytes":1013001186,
         "imagePushedAt":1598625102
      }
   ]
}
1598519828
Total Non-Master Images to delete:206
{
   "imageDetails":[
      {
         "registryId":"754256621582",
         "repositoryName":"correspondence/track-a-query-ecr",
         "imageDigest":"sha256:0acc18ffe0463e932570460c1c7dd451771bd878cc0dd9ae3186adda8132c555",
         "imageTags":[
            "cts-CT-3025-basic-auth-qa-b01b4232"
         ],
         "imageSizeInBytes":1013289296,
         "imagePushedAt":1598478479
      },      
      {
         "registryId":"754256621582",
         "repositoryName":"correspondence/track-a-query-ecr",
         "imageDigest":"sha256:403debe32b1d30c339a78b9570bb1f79a94ae15b80454bba31b2a3f68d7317c0",
         "imageTags":[
            "cts-ct-2900-reindex-command-more-options-7621e533"
         ],
         "imageSizeInBytes":1013249053,
         "imagePushedAt":1592411394
      }
   ]
}"
Deleting Images now...
Images before clean":387
Master Image deletion complete
Non-master images deletion complete
Images after clean:387
Job done.

### CODE REVIEW

**Comment by Polsonby** [comment]
Looks good - can you add to PR explanation of where these untagged images come from? Presumably they're not just left over from our testing but created on an ongoing basis?

**Review by Polsonby — COMMENTED** [review]
LGTM :) 👍 nice one+1

**Review by Polsonby — APPROVED** [review]

**Comment by digitalali-moj** [comment]
will reopen once in play

File: config/jobs/delete-old-ecr-images.sh [hunk 1]
```diff
@@ -0,0 +1,68 @@
+# This script DELETES X old images on the ECR for this repo,
```
Comment (digitalali-moj) [inline config/jobs/delete-old-ecr-images.sh:1]: Note: This will be moved back into the Cron job, just for testing

**Review by Polsonby — COMMENTED** [review]
This looks really nice work 👍 wondering whether you considered making this into a shell script and calling it from the cron thingy? Might be easier for us to play around with it within Circle container or one of our own pods, I'd love to try it out interactively in the context it will be running in.

**Review by Polsonby — APPROVED** [review]
Fab 👍 great job

**Pull request merged**